src/walk.rs

```diff
@@ -329,18 +212,123 @@ impl<W: Write> ReceiverBuffer<W> {
+        let first_path = &self.paths[0];
+        let config = self.config.as_ref();
+
+        let mut builder = OverrideBuilder::new(first_path);
+
+        for pattern in &config.exclude_patterns {
+            builder
+                .add(pattern)
+                .map_err(|e| anyhow!("Malformed exclude pattern: {}", e))?;
+        }
+
+        if config.read_vcsignore {
+            builder.add("!.git/").expect("Invalid exclude pattern");
+        }
+
+        builder
+            .build()
+            .map_err(|_| anyhow!("Mismatch in exclude patterns"))
+    }
+
+    fn build_walker(&self) -> Result<WalkParallel> {
+        let first_path = &self.paths[0];
+        let config = self.config.as_ref();
+
+        let mut builder = WalkBuilder::new(first_path);
+        builder
+            .hidden(config.ignore_hidden)
+            .ignore(config.read_fdignore)
+            .parents(config.read_parent_ignore && (config.read_fdignore || config.read_vcsignore))
+            .git_ignore(config.read_vcsignore)
+            .git_global(config.read_vcsignore)
+            .git_exclude(config.read_vcsignore)
+            .require_git(config.require_git_to_read_vcsignore)
+            .overrides(self.build_overrides()?)
+            .follow_links(config.follow_links)
+            // No need to check for supported platforms, option is unavailable on unsupported ones
+            .same_file_system(config.one_file_system)
+            .max_depth(config.max_depth);
+
+        if config.read_fdignore {
+            builder.add_custom_ignore_filename(".fdignore");
+        }
+
+        if config.read_global_ignore {
+            if let Ok(basedirs) = etcetera::choose_base_strategy() {
+                let global_ignore_file = basedirs.config_dir().join("fd").join("ignore");
+                if global_ignore_file.is_file() {
+                    let result = builder.add_ignore(global_ignore_file);
+                    match result {
+                        Some(ignore::Error::Partial(_)) => (),
+                        Some(err) => {
+                            print_error(format!(
+                                "Malformed pattern in global ignore file. {}.",
+                                err
+                            ));
+                        }
+                        None => (),
+                    }
+                }
+            }
+        }
+
+        for ignore_file in &config.ignore_files {
+            let result = builder.add_ignore(ignore_file);
+            match result {
+                Some(ignore::Error::Partial(_)) => (),
+                Some(err) => {
+                    print_error(format!("Malformed pattern in custom ignore file. {}.", err));
+                }
+                None => (),
+            }
+        }
+
+        for path in &self.paths[1..] {
+            builder.add(path);
+        }
+
+        Ok(builder.threads(config.threads).build_parallel())
+    }
+
+    /// Run the receiver thread.
```

Comment:
Maybe we can clarify a bit here: with this change, the receiving side now either (buffers and) prints results from the main thread. Or spawns new threads for job execution.

Reply:
Done in https://github.com/sharkdp/fd/pull/1413/commits/ed6396d9be4e41f07167899863045d083c3c5201